src/dispatch_private.h

```diff
@@ -141,4 +141,16 @@ void qd_dispatch_unregister_entity(qd_dispatch_t *qd, void *impl);
 /** Set the agent */
 void qd_dispatch_set_agent(qd_dispatch_t *qd, void *agent);
 
+/**
```

Comment:
I am wondering if the function should take ownership of the passed in pointer. I feel more comfortable if the function makes its own copy. Just my 2 cents.